home/export_ordered_sets.py

```diff
@@ -0,0 +1,172 @@
+import copy
+import csv
+import io
+from collections.abc import Iterable
+from dataclasses import asdict, astuple, dataclass, fields
+from math import ceil
+
+from django.http import HttpResponse
+from openpyxl.styles import Border, Font, NamedStyle, Side
+from openpyxl.utils import get_column_letter
+from openpyxl.workbook import Workbook
+from openpyxl.worksheet.worksheet import Worksheet
+from wagtail.query import PageQuerySet  # type: ignore  # No typing available
+
+
+@dataclass
+class ExportRow:
+    """
+    All the data for a single row of an ordered content set
+    """
+
+    name: str
+    profile_field: str
+    page_slugs: str
+    page: str | None
+    unit: str | None
+    before_or_after: str | None
+    contact_field: str | None
+
+    @classmethod
+    def headings(cls) -> list[str]:
+        """
+        The field names, used to write the header rows in exports
+        """
+        return [f.name for f in fields(cls)]
+
+    def to_dict(self) -> dict[str, str]:
+        return asdict(self)
+
+    def to_tuple(self) -> tuple[str]:
+        return astuple(self)
+
+
+class OrderedSetExporter:
+    rows: list[ExportRow]
+
+    def __init__(self, queryset: PageQuerySet):
```

Comment:
I don't think this is a PageQuerySet, since this is a query set of ordered content sets.

Reply:
Should I create a custom query set for these ordered sets?

Reply:
I don't think we have proper typing for django, but could you just use the base `QuerySet` as a return type?